rasa/core/featurizers/tracker_featurizers.py

```diff
@@ -502,7 +502,7 @@ class FullDialogueTrackerFeaturizer(TrackerFeaturizer):
     Creates training data that uses each time output for prediction.
     """
 
-    def training_states_labels_and_entities(
+    def training_states_labels_and_entities(  # type: ignore[override]
```

Comment:
is the return type actually accurate? `trackers_as_actions` can only be `List[Text]`, right?

Reply:
Technically, the `action_name` and `action_text` attributes of `ActionExecuted` event are annotated as `Optional[Text]` in `events.py`, which is probably why the return of `trackers_as_actions` is `List[List[Optional[Text]]]`. Similarly, I could amend the superclass?

Reply:
if there is a way to annotate for the presence of either `action_name` or `action_text` (exclusively), I think that'd be better yes